assignmentPR/MathFunc1.c

```diff
@@ -0,0 +1,316 @@
+        num[iterator] = numString[iterator];
+
+        // Skip all whitespace 
+        if (num[iterator] <= 32)
+        {
+            continue; 
+        }
+
+        // Catch all letters 
+        if (!isNumericChar(num[iterator]) && numString[iterator] != '.')
+        {
+            return false;
+        }
+
+        // if we hit the delimeter 
+        if(numString[iterator] == '.')
+        {
+            num[iterator] = '\0';
+            *c = myAtoi(num);
+
+            return true;
+        }
+    }
+
+    // If the num string contains non-numeric vals, return false 
+    return false;
+}
+
+bool mantissa(char numString[], int* numerator, int *denominator)
+{
+    // If we can get data up to a decimal point, that means we have a characteristic & mantissa 
+       // Get data from the . to the \0 and if we can, return true
+       // save the numerator and denominator
+
+    int negativeFlag = 0;
+    char num[CHAR_ARRAY_SIZE_MAX];
+    cleanCharArray(num, CHAR_ARRAY_SIZE_MAX);
+
+    if (numString[0] == '-' && numString[1] == '0')
+    {
+        negativeFlag = 1;
+    }
+
+    int iterator;
+    for (iterator = 0; numString[iterator] != '\0'; iterator++)
+    {
+        num[iterator] = numString[iterator];
+
+        // Skip all whitespace
+        if (num[iterator] <= WHITESPACE)
+        {
+            continue;
+        }
+
+        // Catch all letters
+        if (!isNumericChar(num[iterator]) && numString[iterator] != '.')
+        {
+            return false;
+        }
+
+        // if we hit the delimeter
+        if (numString[iterator] == '.')
+        {
+            numString[iterator++];
+            cleanCharArray(num, CHAR_ARRAY_SIZE_MAX);
+            break;            
+        }
+    }
+
+    int newNumIterator = 0;
+    for (iterator; numString[iterator] != '\0'; iterator++)
+    {
+        num[newNumIterator] = numString[iterator];
+
+        // Skip all whitespace
+        if (num[newNumIterator] <= WHITESPACE)
+        {
+            continue;
+        }
+
+        // Catch all letters
+        if (!isNumericChar(num[newNumIterator]))
+        {
+            return false;
+        }
+
+        newNumIterator++;
+    }
+
+    // If the num string contains non-numeric vals, return false
+    *numerator = myAtoi(num);
+    if ( negativeFlag )
+    {
+        *numerator *= -1;
+    }
+
+    *denominator = tenToThePowerOf(myStrLen(num));
+
+    return true; 
+}
+
+
+/* Utility Functions */
+
+/* compares two strings, returns 0 for same, -1 for different.
+   method adapted from online source:
+   https://en.wikibooks.org/wiki/C_Programming/string.h/strcmp
+*/
+int compareString(const char *stringOne, const char *stringTwo)
+{
+    for (; *stringOne == *stringTwo; ++stringOne, ++stringTwo)
+    {
+        if (*stringOne == 0)
+        {
+            return 0;
+        }
+    }
+    return *(unsigned char *)
+                       stringOne < *(unsigned char *)stringTwo
+               ? -1
+               : 1;
```

Comment:
I get what's going on in this block (197-200), but you might want to leave a small comment detailing your thought process here just for clarity!